pennylane/drawer/tape_mpl.py

```diff
@@ -191,17 +191,20 @@ def _get_measured_wires(measurements, wires) -> set:
 
 
 def _add_classical_wires(drawer, layers, wires):
-    for cwire, (cwire_layers, layer_wires) in enumerate(zip(layers, wires), start=drawer.n_wires):
-        xs, ys = [], []
 
-        len_diff = len(cwire_layers) - len(layer_wires)
-        if len_diff > 0:
-            layer_wires += [cwire] * len_diff
-        for l, w in zip(cwire_layers, layer_wires):
-            xs.extend([l, l, l])
-            ys.extend([cwire, w, cwire])
+    for cwire, cwire_stretch in layers.items():
+        wire_stretch = wires[cwire]
+        for cwire_layers, layer_wires in zip(cwire_stretch, wire_stretch, strict=True):
+            xs, ys = [], []
 
-        drawer.classical_wire(xs, ys)
+            len_diff = len(cwire_layers) - len(layer_wires)
+            if len_diff > 0:
+                layer_wires += [cwire + drawer.n_wires] * len_diff
+            for l, w in zip(cwire_layers, layer_wires, strict=True):
+                xs.extend([l, l, l])
+                ys.extend([cwire + drawer.n_wires, w, cwire + drawer.n_wires])
+
+            drawer.classical_wire(xs, ys)
```

Comment:
This change is simpler than it looks: Instead of iterating over lists `layers` and `wires`, we now iterate over lists of lists, which in addition are values in dictionaries `layers` and `wires`. This implies that we need an additional indentation level, and instead of `enumerate`, we directly use the keys of the dictionaries.